GUIDELINES.md

```diff
@@ -0,0 +1,159 @@
+            continue
+        sum_values(item_collection)
+```
+
+* Put a space before starting block comments `# like this`, `#not this`
+* Inline comments need two spaces before them `a = 2  # like this`
+* Keep lines to 88 (officially 79) characters or less. You can achieve this by utilising other parts of this guideline, particularly with regards to creating modular code. Splitting over multiple lines is, of course, permissible so long as it doesn't conflict with legibility.
+* When declaring default values, never put spaces around operators like `=`, i.e `def this_is_ok(param=1)`, `def this_is_NOT_ok(param = 1)`. Otherwise, all operators must always have a single space on either side.
+
+
+## Comments and docs
+
+### Mandatory
+
+* At least *a basic docstring* is required for every function/method and class
+* Full, explanatory docstrings are required for all function/methods and classes *if* it will be used in the main body of a code routine.
+* Use Google-style: https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html
+* If you are using type hints then then you can write e.g. `my_arg (str): description` as simply `my_arg: description`.
+
+
+### Encouraged
+
+* Don't state the obvious in comments
+* Before writing a comment, consider whether that information would be better be encoded in a useful variable or function name.
+
+## Workflow
+
+This builds on a much greater body of work, laid out in [nestauk/DAP_playbook](https://github.com/nestauk/DAP_playbook/tree/dev/github). For avoidance of doubt, branches must be linked to a GitHub issue and named accordingly:
+
+```bash
+{GitHub issue number}_{tinyLowerCamelDescription}
+```
+
+For example `14_readme`, which indicates that [this branch](https://github.com/nestauk/ojd_daps/pull/24) refered to [this issue](https://github.com/nestauk/ojd_daps/issues/14).
+
+You should generally favour having a `dev` branch, in addition to your `main` (`master`) branch. Never commit to `dev`, `master` or `main`. 
+
+Only pull requests from branches named `{GitHub issue number}_{tinyLowerCamelDescription}` should ever be accepted.
+
+Please make all PRs and issues reasonably small: they should be trying to achieve roughly one task. Inevitably some simple tasks spawn large numbers of utils, and sometimes these detract from the original PR. In this case, you should stack an new PR on top of your "base" PR, for example as `{GitHub issue number}_{differentLowerCamelDescription}`. In this case the PR / Git merging tree will look like:
+
+    dev <-- 123_originalThing <-- 423_differentThing <-- 578_anotherDifferentThing
+    
+We can then merge the PR `123_originalThing` into `dev`, then `423_differentThing` into `dev` (after calling `git merge dev` on `423_differentThing`), etc until the chain is merged entirely. The nominated reviewer should review the entire chain, before the merge can go ahead. PRs should only be merged if all tests and a review has been signed off.
```

Comment:
I have to admit I'm a little inclear on this workflow. If I'm halfway through working on `123_originalThing` but then realise I need a new feature, which I start to build in `423_differentThing`, how do I then continue work on `123_originalThing` without merging `423_differentThing` back into it?

Reply:
I guess it's still the same model:

`dev` <-- `123_newIdea` <-- `123_originalThing` <-- `423_differentThing`

Reply:
(i.e. just a rebase)